omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java

```diff
@@ -125,6 +129,9 @@ public int compare(SimpleObject o1, SimpleObject o2) {
 	}
 	
 	private SimpleObject parseObs(Obs obs, Locale locale) {
+        if(obs.getConcept() != null && obs.getConcept().isNumeric()){
+            obs.setConcept(this.conceptService.getConceptNumeric(obs.getConcept().getConceptId()));
+        }
```

Comment:
@mogoodrich this is the actual change, we make sure that concepts are propagated as `ConceptNumeric` in the obs representations. This helps kick in some concept numeric specific logic down the line.

Reply:
That's bizarre... so on Obs the concept is of class Concept even when it's numeric, so we reload the concept to cast it back to ConceptNumeric?

Reply:
@mogoodrich indeed. Actually if we had the time, I wanted @JFFigueiredo to investigate a little bit where/why this cast does not happen already up the chain.

It could be that this is because we access the obs through the encounter (so through `EncounterService` and not through `ObsService`).

@Ruhanga maybe you could spend 30 min on this to clarify what happens? As in:
* Write a unit test (or debug an existing unit test) that fetches an obs numeric via its encompassing encounter, and check whether the obs' concept is cast or not into `ConceptNumeric`.

Reply:
Ok...

Reply:
If we don't figure it out and merge this change as-is, let's add a comment to the code saying we aren't quite sure why we have to do this? :)

Reply:
I've tried to reproduce the issue through [this commit](https://github.com/openmrs/openmrs-core/commit/b2794c526565f7cbe1ed0983c6ae9ac1cb3df7dd) but seem not to get exactly to problem for now. I'll take time over the weekend to investigate further.

Reply:
@mogoodrich I have asked @JFFigueiredo to give simple steps to reproduce the issue on QA Ref App.

In short, if a concept numeric disallows decimals, then everything should work already and 'Show Details' should not display any decimals. However this isn't the case, precisely because of the `ConceptNumeric` casting not happening issue.

If everything takes too much time to investigate, we will merge the current PR with a "no idea why we have to do this" comment as you suggested :-)

Reply:
@mks-d , @mogoodrich I added the steps to reproduce the bug on this ticket [RA-1945](https://issues.openmrs.org/browse/RA-1945).
Here is a quick view of the problem, even when allowDecimals not enabled, it add the ".0" on "Show Details"
![integer](https://user-images.githubusercontent.com/72515287/127143832-e3339bf5-b094-4d56-b367-30d64be4e4dc.gif)